### PR TITLE
Add exception handler

### PR DESCRIPTION
**Pull request by k0kubun**
## Background
- We want to handle exception with `Raven.capture_exception`.
  - There may be some people who don't want to use `Raven`.
- Exceptions in `Barbeque::Worker` must be rescued and given to this exception handler. 
  - Other exception are properly handled if we configure raven.

## Changes
Add exception handler to handle exception with `Rails.logger` or `Raven`.

Please review :eyeglasses: @cookpad/dev-infra

### CODE REVIEW

**Comment by k0kubun** [comment]
To keep barbeque.gem simple, I changed `Barbeque::ExceptionHandler` to be pluggable.

**Comment by eisuke** [comment]
pluggble 👍 
LGTM

File: lib/barbeque/exception_handler.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+require 'barbeque/configuration'
+
+module Barbeque
+  module ExceptionHandler
+    def self.handle_exception(e)
+      handler = const_get(Barbeque.config.exception_handler, false)
```
Comment (makimoto) [inline lib/barbeque/exception_handler.rb]: IMO, `const_get` takes a cost so you might like to memorize `handler` like `@handler ||= const_get(Barbeque.config.exception_handler, false)`.

Comment (k0kubun) [inline lib/barbeque/exception_handler.rb]: Fixed https://github.com/cookpad/barbeque/pull/7/commits/b4ea76e636881ca021481969f07cac1aad771256

**Comment by makimoto** [comment]
LGTM

**Pull request merged**